ConsoleSession.java

```diff
@@ -280,7 +288,7 @@ public final void close() {
         session.close();
         client.close();
         try {
-            historyFile.flush();
+            history.flush();
```

Comment:
Given that we only flush once I think you can change the field `history` to be `History` rather than `PersistentHistory` and then in this `try-catch` you can just check `if(history instanceof PeristentHistory) { history.flush(); }` , so that we don't need to create a fake empty class.